lib/children/carelinkLoad.js

```diff
@@ -107,23 +103,45 @@ function fetch(groupId, filename, config, cb) {
         return;
       }
 
-      dataStream.on('data', function (chunk) {
+      var updates = {};
+
+      var bytesPulled = 0;
+      var recordData = '';
+      var recordRegex = /DEVICE DATA \((\d+?) records\)/i;
+      var recordCount = 0;
+
+      dataStream.on('data', function(chunk) {
         if (bytesPulled === 0) {
-          log.info('First byte pulled in [%s] millis.', new Date().valueOf() - startTime);
+          log.info('First byte pulled from %s in %sms', type, new Date().valueOf() - startTime);
         }
         bytesPulled += chunk.length;
+
+        // Look for record count
+        if (recordData != null) {
+          recordData += chunk.toString();
+          var match = recordRegex.exec(recordData);
+          if (match && match.length) {
+            recordCount = parseInt(match[1]);
+            recordData = null;
+          }
+        }
       });
-      dataStream.on('end', function () {
-        log.info('%s data pulled in [%s] millis.  Size [%s]', type, new Date().valueOf() - startTime, bytesPulled);
+      dataStream.on('end', function() {
+        log.info('All %s bytes pulled from %s in %sms', bytesPulled, type, new Date().valueOf() - startTime);
+
+        // If no records, then error (backwards compatible)
+        if (recordCount === 0) {
```

Comment:
Is a 204 always treated as an error ? I mean its legit as there is ``no content`` but is it classed as an error in general